src/prefect/deployments.py

```diff
@@ -41,7 +41,7 @@
 @sync_compatible
 @inject_client
 async def run_deployment(
-    name: str,
+    name_or_id: str,
```

Comment:
We cannot change the name of the parameter, as that is a breaking change. Instead we should update the docstring to reflect that this can be an id.